Code/src/pipeline/gnn_pipeline.py

```diff
@@ -0,0 +1,286 @@
+        Checker.graph_data_split_to_train_val_test(self)
+
+        # TODO: might be able to move this to somewhere more meaningful,
+        # but it is needed here at the latest
+        self.device = "cuda" if torch.cuda.is_available() else "cpu"
+
+        self.train_loader = LinkNeighborLoader(
+            data=self.train_data,
+            edge_label_index=self.edge_index[:, self.train_indices],
+            edge_label=self.y[self.train_indices],
+            batch_size=batch_size,
+            num_neighbors=num_neighbors,
+            shuffle=True,
+        )
+
+        self.val_loader = LinkNeighborLoader(
+            data=self.val_data,
+            edge_label_index=self.edge_index[:, self.val_indices],
+            edge_label=self.y[self.val_indices],
+            batch_size=batch_size,
+            num_neighbors=num_neighbors,
+            shuffle=False,
+        )
+
+        self.test_loader = LinkNeighborLoader(
+            data=self.test_data,
+            edge_label_index=self.edge_index[:, self.test_indices],
+            edge_label=self.y[self.test_indices],
+            batch_size=batch_size,
+            num_neighbors=num_neighbors,
+            shuffle=False,
+        )
+
+        self.preprocessed["got_data_loaders"] = True
+                
+    def run_model_specific_steps(self, edge_features=list[str], edge_features_to_scale= list[str], num_neighbors=[100,100], batch_size=8192):
+        self.split_train_test_val_graph(edge_features)
+        self.scale_edge_features(edge_features_to_scale)
+        self.get_data_loaders(num_neighbors, batch_size)
+        
+    def initialize_training(
+        self,
+        threshold: float=0.5,
+        epochs: int=50,
+        patience: int=10,
+    ) -> None:
+        """Setup the model pipeline for training: metrics, model,
+        optimizer, scheduler, and criterion
+        """
+        self.threshold = threshold
+        self.epochs = epochs
+        self.patience = patience
+        
+        
+
+        # Since `initialize_training` is run after preprocessing is
+        # done, we can define the node and edge features here. This
+        # does assume that column ordering between data frames and
+        # tensors is preserved, and it removes node and edge id
+        # TODO: ran into issue with this line bc node_id has already been dropped
+        if 'node_id' in self.nodes.columns:
+            self.node_feature_labels = self.nodes.drop(columns="node_id").columns
+        else:
+            self.node_feature_labels = self.nodes.columns
+        self.edge_feature_labels = self.df[self.edge_features].drop(columns="edge_id").columns
+
+
+
+        # Model setup
+        num_edge_features = self.train_data.edge_attr.shape[1]-1  # num edge feats - edge_id
+        num_node_features = self.train_data.x.shape[1]
+        self.model = GINe(n_node_feats=num_node_features, n_edge_feats=num_edge_features).to(self.device)
+        self.trainer = GNNTrainer(self.model, self)
+
+
```

Comment:
```suggestion
        self.threshold = threshold
        self.epochs = epochs
        self.patience = patience

        # Since `initialize_training` is run after preprocessing is
        # done, we can define the node and edge features here. This
        # does assume that column ordering between data frames and
        # tensors is preserved, and it removes node and edge id
        # TODO: ran into issue with this line bc node_id has already been dropped
        if 'node_id' in self.nodes.columns:
            self.node_feature_labels = self.nodes.drop(columns="node_id").columns
        else:
            self.node_feature_labels = self.nodes.columns
        self.edge_feature_labels = self.df[self.edge_features].drop(columns="edge_id").columns

        # Model setup
        num_edge_features = self.train_data.edge_attr.shape[1]-1  # num edge feats - edge_id
        num_node_features = self.train_data.x.shape[1]
        self.model = GINe(n_node_feats=num_node_features, n_edge_feats=num_edge_features).to(self.device)
        self.trainer = GNNTrainer(self.model, self)

```

Just some extra line breaks